auth/server/src/main/java/com/seagates3/controller/AccountController.java

```diff
@@ -171,6 +190,22 @@ public ServerResponse create() {
           LOGGER.error("Create account delay failing - ", e);
           Thread.currentThread().interrupt();
         }
+        // Handle multi-threaded create API calls by deleting extra accounts,
+        // if account limit creation exceeded due to multiple thread/multiple
+        // node create() API calls.
+        try {
+          accountCount = accountDao.getTotalCountOfAccounts();
+
+          if (accountCount > maxAllowedLdapResults) {
+            // delete newly created account since we exceeded account
+            // creation limit.
+            accountDao.ldap_delete_account(account);
+            return accountResponseGenerator.internalServerError();
+          }
+        }
+        catch (DataAccessException ex) {
```

Comment:
can you catch exception separately for getTotalCountOfAccounts and ldap_delete_account to differentiate what failed

Reply:
done